pkg/catalog/service.go

```diff
@@ -146,19 +146,21 @@ func (mc *MeshCatalog) GetTargetPortToProtocolMappingForService(svc service.Mesh
 func (mc *MeshCatalog) GetPortToProtocolMappingForService(svc service.MeshService) (map[uint32]string, error) {
 	portToProtocolMap := make(map[uint32]string)
 
-	k8sSvc := mc.kubeController.GetService(svc)
-	if k8sSvc == nil {
-		return nil, errors.Wrapf(ErrServiceNotFound, "Error retrieving k8s service %s", svc)
+	for _, provider := range mc.serviceProviders {
+		currentPortToProtocolMap, err := provider.GetTargetPortToProtocolMappingForService(svc)
+		if err != nil {
+			return nil, err
+		}
+		for key, value := range currentPortToProtocolMap {
+			if v, ok := portToProtocolMap[key]; ok && v != value {
+				return nil, errors.Errorf("Error fetching port:protocol, multiple entries found on same port for service %s", svc)
```

Comment:
Clarify log to something like:
```suggestion
				return nil, errors.Errorf("Unexpected protocol %s for port %d, expected protocol %s, service: %s, provider: %s", value, key, v, svc, provider.GetID())
```